x-pack/plugins/maps/server/routes.js

```diff
@@ -26,53 +27,94 @@ export function initRoutes(server, licenseUid) {
 
   server.route({
     method: 'GET',
-    path: `${ROOT}/data/ems`,
+    path: `${ROOT}/${EMS_DATA_FILE_PATH}`,
     handler: async (request) => {
 
       if (!request.query.id) {
         server.log('warning', 'Must supply id parameters to retrieve EMS file');
         return null;
       }
 
-      const ems = await getEMSResources(licenseUid);
-
+      const ems = await getEMSResources(emsClient, mapConfig.includeElasticMapsService, licenseUid, true);
       const layer = ems.fileLayers.find(layer => layer.id === request.query.id);
       if (!layer) {
         return null;
       }
 
-      const file = await fetch(layer.url);
-      return await file.json();
+      try {
+        const file = await fetch(layer.url);
+        return await file.json();
+      } catch(e) {
+        server.log('warning', `Cannot connect to EMS for file, error: ${e.message}`);
+        throw Boom.badRequest(`Cannot connect to EMS`);
+      }
+
+    }
+  });
+
+
+  server.route({
+    method: 'GET',
+    path: `${ROOT}/${EMS_DATA_TMS_PATH}`,
+    handler: async (request, h) => {
+
+      if (!request.query.id ||
+        typeof parseInt(request.query.x, 10) !== 'number' ||
+        typeof parseInt(request.query.y, 10) !== 'number' ||
+        typeof parseInt(request.query.z, 10) !== 'number'
+      ) {
+        server.log('warning', 'Must supply id/x/y/z parameters to retrieve EMS tile');
+        return null;
+      }
+
+      const ems = await getEMSResources(emsClient, mapConfig.includeElasticMapsService, licenseUid, true);
+      const tmsService = ems.tmsServices.find(layer => layer.id === request.query.id);
+      if (!tmsService) {
+        return null;
+      }
+
+      const url = tmsService.url
+        .replace('{x}', request.query.x)
+        .replace('{y}', request.query.y)
+        .replace('{z}', request.query.z);
 
+      try {
+        const tile = await fetch(url);
+        const arrayBuffer = await tile.arrayBuffer();
+        const buffer = Buffer.from(arrayBuffer);
+        let response = h.response(buffer);
+        response = response.bytes(buffer.length);
+        response = response.header('Content-Disposition', 'inline');
+        response = response.header('Content-type', 'image/png');
+        response = response.encoding('binary');
+        return response;
+      } catch(e) {
+        server.log('warning', `Cannot connect to EMS for tile, error: ${e.message}`);
+        throw Boom.badRequest(`Cannot connect to EMS`);
+      }
     }
   });
 
   server.route({
     method: 'GET',
-    path: `${ROOT}/meta`,
+    path: `${ROOT}/${EMS_META_PATH}`,
     handler: async () => {
 
       let ems;
       try {
-        ems = await getEMSResources(licenseUid);
+        ems = await getEMSResources(emsClient, mapConfig.includeElasticMapsService, licenseUid, !mapConfig.proxyElasticMapsServiceInMaps);
```

Comment:
`!mapConfig.proxyElasticMapsServiceInMaps` is not right and should be `!mapConfig.proxyElasticMapsServiceInMaps` or just `true`

Reply:
hmmm...  maybe throw if `proxyElasticMapsServiceInMaps` is false? not sure if this meta call should be functional at all if the proxy-ing is turned off.

Reply:
returning 404 now